test/infrastructure/docker/README.md

```diff
@@ -2,6 +2,8 @@
 
 CAPD is a reference implementation of an infrastructure provider for the Cluster API project using Docker.
 
+**NOTE:** The Docker provider is **not** designed for production use and is intended for development environments only.
```

Comment:
nit. IMO CAPD goals are actually providing the intend clearly.
```
CAPD Goals
To be a the reference implementation of an infrastructure provider.
The code is highly trusted and used in testing of ClusterAPI.
This provider can be used as a guide for developers looking to implement their own infrastructure provider.
```

Reply:
Personally, I'm fine if being redundant here, we are adding this sentence consistently across all the docs.

Reply:
I've left this in - please confirm if this is to be removed / blocking pr merge

Reply:
It is ok to leave as it is